### PR TITLE
Update mkdocs

### PR DESCRIPTION
**Pull request by engelmi**
This PR adds to the mkdocs:
- configuration
- architecture diagram for the proxy services

### CODE REVIEW

File: doc/docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+| Load order | hirte | hirte-agent |
+|---|---|---|
+| 1 | `/usr/share/hirte/config/hirte.conf` | `/usr/share/hirte-agent/config/agent.conf` |
+| 2 | `/etc/hirte/hirte.conf` | `/etc/hirte/agent.conf` |
+| 3 | `/etc/hirte/hirte.conf.d` | `/etc/hirte/agent.conf.d` |
+
+Based on the load order, settings from a previously read configuration file will be overridden by subsequent files.
+For example, the default setting for `AllowedNodeNames` in `/usr/share/hirte/config/hirte.conf` is an empty list and
+needs to be overridden by either `/etc/hirte/hirte.conf` or files in `/etc/hirte/hirte.conf.d`. Configuration files are
```
Comment (pypingou) [inline doc/docs/configuration.md]:
```suggestion
can be overridden by either editing `/etc/hirte/hirte.conf` or adding a file in `/etc/hirte/hirte.conf.d`. Configuration files in
```

Just to be sure, if I configure `AllowedNodeNames` in different files in `hirte.conf.d`, only the last one is taken into account, right? It's not combining the different values found, right?

File: doc/docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+sorted alphabetically and read in ascending order.
```
Comment (pypingou) [inline doc/docs/configuration.md]:
```suggestion
 `/etc/hirte/hirte.conf.d` are sorted alphabetically and read in ascending order.
```

File: doc/mkdocs.yml [hunk 3]
```diff
@@ -8,19 +8,18 @@ copyright: © Red Hat
...
   - Getting Started: getting_started.md
+  - Configuration: configuration.md
   - DBus API: 
     - Description: api/description.md
     - Examples: api/examples.md
-  - Proxy Services: 
-    - Concept: proxy/concept.md
-  - Configuration files:
-    - Hirte controller: man/hirte_conf.md
-    - Hirte agent: man/hirte_agent_conf.md
```
Comment (pypingou) [inline doc/mkdocs.yml:18]: I kinda like the configuration files to have their own section, even if that's the man pages indeed.

File: doc/docs/configuration.md (hunk 1 above)
Comment (engelmi) [reply doc/docs/configuration.md]: Yes, that is correct. There were talks about merging, but nothing more.

File: doc/mkdocs.yml (hunk 3 above)
Comment (engelmi) [reply doc/mkdocs.yml:18]: What do you think about 
```
MAN Pages
|- Binaries
   |- hirte
   ...
|- Configuration
   |- hirte
   ...
```

Comment (pypingou) [reply doc/mkdocs.yml:18]: ok, why not :)

File: doc/docs/configuration.md (hunk 1 above)
Comment (engelmi) [reply doc/docs/configuration.md]: Changed.

File: doc/mkdocs.yml (hunk 3 above)
Comment (engelmi) [reply doc/mkdocs.yml:18]: Changed, thanks!

**Review by pypingou — APPROVED** [review]
Get it in!

**Pull request merged**